### PR TITLE
Uppdated the model to output track and a few other datapoints.

### PR DESCRIPTION
**Pull request by Fberbari**
The model now produces everything sensor fusion would produce, so we've got a full software-in-the-loop simulation now.

### CODE REVIEW

**Comment by Fberbari** [comment]
Note, I gotta do a final test or 2 before I'm ready for merge. I put this up so the review could begin, but I'm realizing there's not much to review... sigh.

File: Infinion/ZeroPilotSwHooks/SensorFusionIntercept.cpp
Comment (NixonCTChan) [inline Infinion/ZeroPilotSwHooks/SensorFusionIntercept.cpp:13]: Are we gonna do anything about these declarations of Pi anytime soon?

**Review by NixonCTChan — COMMENTED** [review]
Everything looks pretty good! Can't really review the simulink model, but all the other changes seem ok!

File: Infinion/ZeroPilotSwHooks/SensorFusionIntercept.cpp
Comment (Fberbari) [reply Infinion/ZeroPilotSwHooks/SensorFusionIntercept.cpp:13]: yes, I'm gonna treat that as my next task, but I'm gonna be making a dedicated PR for it.

File: Infinion/GeneratedCCode/Headers/simulink/include/rtw_solver.h
Comment (Pop0097) [inline Infinion/GeneratedCCode/Headers/simulink/include/rtw_solver.h:120]: Pretty cool how you're using general data types. Will need to learn more about them

File: Infinion/ZeroPilotSwHooks/SensorFusionIntercept.cpp
Comment (Pop0097) [inline Infinion/ZeroPilotSwHooks/SensorFusionIntercept.cpp:100]: chuckles, Infinion is in danger

File: Infinion/ZeroPilotSwHooks/SensorFusionIntercept.cpp
Comment (Pop0097) [inline Infinion/ZeroPilotSwHooks/SensorFusionIntercept.cpp:149]: This do while loop seems a bit suspicious to me:

with the way it's currently structured, it appears as if we will go straight to the end of our file and return the last entry. 

Am I misunderstanding this? 

http://www.cplusplus.com/reference/string/string/empty/

**Review by Pop0097 — COMMENTED** [review]
It seems good to me overall. Just one thing I am slightly unsure of.

File: Infinion/GeneratedCCode/Headers/simulink/include/rtw_solver.h
Comment (Fberbari) [reply Infinion/GeneratedCCode/Headers/simulink/include/rtw_solver.h:120]: not my doing lol, all generated. but yeah, quite creative

File: Infinion/ZeroPilotSwHooks/SensorFusionIntercept.cpp
Comment (Fberbari) [reply Infinion/ZeroPilotSwHooks/SensorFusionIntercept.cpp:149]: the getline function saves its state. It always starts at the first line and iterates until it reaches the end. 

This was thoroughly tested in past versions of the simulation, so I can be sure it works.

Comment (Pop0097) [reply Infinion/ZeroPilotSwHooks/SensorFusionIntercept.cpp:149]: Okay, then you have my approval

Comment (Fberbari) [reply Infinion/ZeroPilotSwHooks/SensorFusionIntercept.cpp:149]: thank you sir!

**Review by Pop0097 — APPROVED** [review]
Nice

**Pull request merged**
